packages/react-topology/src/pipelines/components/nodes/TaskNode.tsx

```diff
@@ -249,114 +270,152 @@ const TaskNode: React.FC<TaskNodeProps> = ({
     />
   );
 
-  const taskPill = (
-    <g
-      className={pillClasses}
-      transform={`translate(${whenOffset + whenSize}, 0)`}
-      ref={hoverRef}
-      onClick={onSelect}
-      onContextMenu={onContextMenu}
-    >
-      <NodeShadows />
-      <rect
-        x={0}
-        y={0}
-        width={pillWidth}
-        height={height}
-        rx={height / 2}
-        className={css(styles.topologyPipelinesPillBackground)}
-        filter={filter}
-      />
-      <g transform={`translate(${textStartX}, ${paddingY + textHeight / 2 + 1})`}>
-        {element.getLabel() !== label && !disableTooltip ? (
-          <Tooltip content={element.getLabel()}>
-            <g>{nameLabel}</g>
-          </Tooltip>
-        ) : (
-          nameLabel
-        )}
-      </g>
-      {status && showStatusState && (
-        <g transform={`translate(${statusStartX + paddingX / 2}, ${(height - statusIconSize) / 2})`} ref={statusRef}>
-          <g
+  const renderTask = () => {
+    if (showStatusState && !scaleNode && hideDetailsAtMedium && detailsLevel !== ScaleDetailsLevel.high) {
+      const statusBackgroundRadius = statusIconSize / 2 + 4;
+      const height = element.getBounds().height;
+      const upScale = 1 / scale;
+
+      return (
+        <g transform={`translate(0, ${(height - statusBackgroundRadius * 2 * upScale) / 2}) scale(${upScale})`}>
+          <circle
             className={css(
-              styles.topologyPipelinesPillStatus,
+              styles.topologyPipelinesStatusIconBackground,
```

Comment:
In pf-react-core we consider removing classes breaking incase consumers are targeting them for any customization.  Would this impact topology consumers?

Reply:
I can leave it as it has no effect.